t/api/14-webuiplugin.t

```diff
@@ -0,0 +1,75 @@
+# Copyright (C) 2019 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+BEGIN {
+    unshift @INC, 'lib';
+}
+
+use Mojo::Base -strict;
+use Mojo::File qw(tempdir);
+use FindBin;
+use lib "$FindBin::Bin/../lib";
+use Test::More;
+use Test::Mojo;
+use OpenQA::Test::Database;
+# use Test::Warnings;
+
+my $test_openapi = eval " use Mojolicious::Plugin::OpenAPI; 1; ";
+plan skip_all => 'OpenAPI plugin is not installed, skipping' unless (defined $test_openapi);
+
+my $tempdir = tempdir;
+$ENV{OPENQA_CONFIG} = $tempdir;
+
+# we must create db, otherwise OpenQA::WebAPI will try to
+my $db = OpenQA::Test::Database->new->create(skip_fixtures => 1);
+
+sub check_plugin {
+    my ($config, $expected_status) = @_;
+
+    $tempdir->child("openqa.ini")->spurt($config);
+    my $t = eval { Test::Mojo->new('OpenQA::WebAPI'); };
+    # our $last_warning = '';
+    # $SIG{__WARN__} = sub { $last_warning = shift; };
+
+    if (defined $t) {
+        my $app = $t->app;
+        $t->ua(OpenQA::Client->new()->ioloop(Mojo::IOLoop->singleton));
+        $t->app($app);
+        $t->ua->apikey('PERCIVALKEY01');
+        $t->ua->apisecret('PERCIVALSECRET01');
+    }
+    ok(defined $t || !defined $expected_status, "WebAPI startup");
+    ok(!defined $t || $t->get_ok('/plugin/hello_world')->status_is($expected_status), "API");
+}
```

Comment:
This function looks like a temporary hack, so i'll just assume the `eval` and stuff will get cleaned up later. 😉

Reply:
The idea here is to make sure that app doesn't start when `required` plugin fails to load. Thus it does `eval` and then checks if expected error really did happen. 
The lines around $last_warning are needed to make sure that test passes when warning checks are enabled because otherwise warning leads to test failure: `coerce(1) will be deprecated. at /usr/lib/perl5/vendor_perl/5.28.1/Mojolicious/Plugin/OpenAPI.pm line 29`
The dance around apikey() I took from another test to make sure that privileged user authorizes. (Not sure how to make it work otherwise).